deploy/cmd/create.go

```diff
@@ -0,0 +1,171 @@
+			if err != nil {
+				return err
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			group, _ := errgroup.WithContext(ctx)
+
+			// Listen to on chain events and send the manifest when required
+			group.Go(func() error {
+				if err = ChainEmitter(ctx, clientCtx, DeploymentDataUpdateHandler(dd), SendManifestHander(clientCtx, dd)); err != nil {
+					log.Error("error watching events", err)
+				}
+				return err
+			})
+
+			// Send the deployment creation transaction
+			group.Go(func() error {
+				if err = TxCreateDeployment(clientCtx, cmd.Flags(), dd); err != nil {
+					log.Error("error creating deployment", err)
+				}
+				return err
+			})
+
+			// Wait for the leases to be created and then start polling the provider for service availability
+			group.Go(func() error {
+				if err = WaitForLeasesAndPollService(clientCtx, dd, cancel); err != nil {
+					log.Error("error listening for service", err)
+				}
+				return err
+			})
+
+			return group.Wait()
+		},
+	}
+
+	cmd.Flags().String(flags.FlagChainID, "", "The network chain ID")
+	if err := viper.BindPFlag(flags.FlagChainID, cmd.Flags().Lookup(flags.FlagChainID)); err != nil {
+		return nil
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	dcli.AddDeploymentIDFlags(cmd.Flags())
+
+	return cmd
+}
+
+// WaitForLeasesAndPollService waits for leases
+func WaitForLeasesAndPollService(clientCtx client.Context, dd *DeploymentData, cancel context.CancelFunc) error {
+	log := logger
+	pclient := pmodule.AppModuleBasic{}.GetQueryClient(clientCtx)
+	timeout := time.After(150 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		case <-timeout:
+			log.Info("timed out (90s) listening for deployment to be available")
+			cancel()
+			return nil
+		case <-tick:
+			if dd.ExpectedLeases() {
+				for _, l := range dd.Leases() {
+
+					var (
+						p   *ptypes.Provider
+						err error
+					)
+					if err := retry.Do(func() error {
+						res, err := pclient.Provider(
+							context.Background(),
+							&ptypes.QueryProviderRequest{Owner: l.Provider},
+						)
+						if err != nil {
+							// TODO: Log retry?
+							return err
+						}
+
+						p = &res.Provider
+						return nil
+					}); err != nil {
+						cancel()
+						return fmt.Errorf("error querying provider: %w", err)
+					}
+
+					// TODO: Move to using service status here?
+					var ls *ctypes.LeaseStatus
+					if err := retry.Do(func() error {
+						ls, err = gateway.NewClient().LeaseStatus(context.Background(), p.HostURI, l)
+						if err != nil {
+							return err
+						}
+						return nil
+					}); err != nil {
+						cancel()
+						return fmt.Errorf("error querying lease status: %w", err)
+					}
+
+					for _, s := range ls.Services {
+						// TODO: Much better logging/ux could be put in here: waiting, timeouts etc...
+						if s.Available == s.Total {
+							log.Info(strings.Join(s.URIs, ","), "name", s.Name, "available", s.Available)
```

Comment:
This looks like it would get logged each time a value is retrieved by `<- tick`. So once ever 0.5 seconds right now.

This should probably be logged once when the service has `s.Available == s.Total` .After that it doesn't need to be logged again.